net/dns/resolver/forwarder.go

```diff
@@ -699,6 +687,17 @@ func (f *forwarder) sendUDP(ctx context.Context, fq *forwardQuery, rr resolverAn
 	return out, nil
 }
 
+func (f *forwarder) dialContext() dnscache.DialContextFunc {
```

Comment:
name the func "dialer"? It returns a dialer, after all.

Reply:
`forwarder` already has `dialer *tsdial.Dialer` as a field. This returns one of its methods, so I named it after the function's type name. But don't mind renaming if there's a better name for either the new method or the existing field.

Reply:
getDialerType?